packages/next/src/views/Edit/index.client.tsx

```diff
@@ -24,19 +25,25 @@ export const EditViewClient: React.FC = () => {
   const { dispatchFormQueryParams } = useFormQueryParams()
 
   const { getComponentMap } = useComponentMap()
+  const params = useSearchParams()
+
+  const locale = params.get('locale')
 
   const { Edit, actionsMap } = getComponentMap({
     collectionSlug,
     globalSlug,
   })
 
   const onSave = useCallback(
+    // eslint-disable-next-line @typescript-eslint/require-await
     async (json: { doc }) => {
       void getVersions()
       void getDocPermissions()
 
       if (!isEditing) {
-        router.push(`${adminRoute}/collections/${collectionSlug}/${json?.doc?.id}`)
+        const baseRedirectRoute = `${adminRoute}/collections/${collectionSlug}/${json?.doc?.id}`
+        const redirectRoute = locale ? baseRedirectRoute + `?locale=${locale}` : baseRedirectRoute
```

Comment:
You could make this a one-liner.
```suggestion
        const redirectRoute = `${adminRoute}/collections/${collectionSlug}/${json?.doc?.id}${locale ? `?locale=${locale}` : ''}`
```